app/src/main/java/org/mozilla/rocket/preference/Preferences.kt

```diff
@@ -0,0 +1,67 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 20; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.rocket.preference
+
+import android.content.SharedPreferences
+
+interface Preferences {
```

Comment:
I like this interface cause we can do unit testing on shared preference.
Just one thing:
Should we expose getString(key, defaultValue) api? or just restrict the caller to use something like `shouldShowOnboarding() : Boolean`

the later give us better control on how callers can use this new Preferences interface.

Reply:
Preferences is more like a low-level interface with get/put abstraction, so that we can have the 
 ability to change the underlying implementation at will. PrivateBrowsingSettings and GlobalSettings, by contrast, provide more high-level methods and also restrict the access to Preferences.